server/balancer.go

```diff
@@ -127,7 +128,7 @@ func (cb *capacityBalancer) ScoreType() scoreType {
 // selectBalanceRegion tries to select a store leader region to do balance and returns true, but if we cannot find any,
 // we try to find a store follower region and returns false.
 func (cb *capacityBalancer) selectBalanceRegion(cluster *clusterInfo, stores []*storeInfo) (*metapb.Region, *metapb.Peer, *metapb.Peer, bool) {
-	store := selectFromStore(stores, cluster.getUnknownStores(), cb.filters, cb.st)
+	store := selectFromStore(stores, nil, cb.filters, cb.st)
```

Comment:
Unknown stores will be filtered out in `stateFilter`, no need to maintain another unknown stores map now.